### PR TITLE
Secret Store Authorization Implementation

### PR DESCRIPTION
**Pull request by openwarrior**
SecretStoreAuthzModule:-
Performs all related work of generating new key and secret pair, Storing them as account,
Retrieving stored account, Deleting stored account

ModuleKeeper:-
This class have sole reason for storing the Authentication module.
So that can be accessed from anywhere in the code at any time.

Session:-
Body to store key, secret and other related stuff

AuthData:-
Contain Strings which are to be used by package.

### CODE REVIEW

File: src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+    @Override
+    public boolean handleError(HttpException exception) {
+        return false;
+    }
+
+    private void addAccount(String data) throws JSONException {
+        JSONObject auth = new JSONObject(data);
+        OAuth2AuthzSession session = new OAuth2AuthzSession();
+        session.setAccountId(AuthData.NAME);
+        session.setAccessToken(auth.getString("key") + ":" + auth.getString("secret"));
```
Comment (jpkrohling) [inline src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java]: Is this how it's supposed to be used? Secret Store's tokens are not OAuth Tokens.

Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java]: @jpkrohling  I was just using that to store key:secret . But right, Its not a good practice. repairing it

Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/auth/SecretStoreAuthzModule.java]: @jpkrohling Changed to customized Session class. :)

**Comment by openwarrior** [comment]
@danielpassos  Please review the pull request. And also ping summers as I can't ping him. Thanks

File: src/main/java/org/hawkular/client/android/auth/AuthData.java [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.client.android.auth;
+
+public final class AuthData {
```
Comment (pilhuhn) [inline src/main/java/org/hawkular/client/android/auth/AuthData.java:23]: Can you please add a class comment about that this class is supposed to do?

Could those constants be moved to `auth.Session` class?

**Comment by pilhuhn** [comment]
So that is basically a re-make of the other PR that you did and then closed to be able to do other work first?

**Comment by openwarrior** [comment]
@pilhuhn Approx. With some little enhancements told by summers and jpkrohling

File: src/main/java/org/hawkular/client/android/auth/AuthData.java (hunk 2 above)
Comment (openwarrior) [inline src/main/java/org/hawkular/client/android/auth/AuthData.java:23]: Adding comment to this class but can't be moved as session is something to be stored in aerogear store but this class will be needed multiple times as a map for auth module :)

**Pull request merged**